src/App.js

```diff
@@ -1,23 +1,85 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from 'react';
+import PageHeader from './components/PageHeader';
+import PageMain from './components/PageMain';
+import PageFooter from './components/PageFooter';
+import SearchForm from './components/SearchForm';
+import MovieDetails from './components/MovieDetails';
+import MoviesList from './components/MoviesList';
+import MoviesModal from './components/MoviesModal';
+import ErrorBoundary from './utils/ErrorBoundary';
+import { ReactComponent as SearchIcon } from './images/icons/search.svg';
+import './styles/App.scss';
 
 function App() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isMovieDetailsMode, setIsMovieDetailsMode] = useState(false);
+  const [movieDetails, setMovieDetails] = useState({ type: 'add', data: {} });
+
+  const showMovieModal = (type, movieData) => {
+    setMovieDetails({ type, data: movieData });
+    setIsModalVisible(true);
+  };
+
+  const handleHeaderBtnClick = () => {
```

Comment:
Please do not name functions as `handleHeaderBtnClick` because it says nothing about what it will do but when it will do. Please use names which will describe what function will do.

Reply:
So for other devs and for you (probably after two weeks) code will be more easier to read because it will be self-documented https://www.sitepoint.com/self-documenting-javascript/

Reply:
Thank you for the helpful link! Fixed